executor/converter/convert_expr.go

```diff
@@ -106,6 +106,8 @@ func (c *expressionConverter) Leave(in ast.Node) (out ast.Node, ok bool) {
 		c.funcCast(v)
 	case *ast.FuncSubstringExpr:
 		c.funcSubstring(v)
+	case *ast.FuncSubstringIndexExpr:
+		c.funcSubstringIndex(v)
 	case *ast.FuncLocateExpr:
 		c.funcLocate(v)
 	case *ast.FuncTrimExpr:
```

Comment:
We can add a case like `case ast.ExprNode` to cover it, but the visiting cost would be much higher, because switch to an interface costs much more than switch to a real type.